src/main/java/com/cst438/controller/StudentController.java

```diff
@@ -57,25 +102,62 @@ public EnrollmentDTO addCourse(
 		    @PathVariable int sectionNo,
             @RequestParam("studentId") int studentId ) {
 
-        // TODO
-
-        // check that the Section entity with primary key sectionNo exists
-        // check that today is between addDate and addDeadline for the section
-        // check that student is not already enrolled into this section
-        // create a new enrollment entity and save.  The enrollment grade will
-        // be NULL until instructor enters final grades for the course.
-
-        // remove the following line when done.
-        return null;
-
+        Optional<Section> sectionOptional = sectionRepository.findById(sectionNo);
+        if (sectionOptional.isEmpty()) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Section not found");
+        }
+
+        Section section = sectionOptional.get();
+        Date currentDate = new Date();
+        if (currentDate.before(section.getTerm().getAddDate()) || currentDate.after(section.getTerm().getAddDeadline())) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Current date is not within the add period");
+        }
+
+        Enrollment existingEnrollment = enrollmentRepository.findEnrollmentBySectionNoAndStudentId(sectionNo, studentId);
```

Comment:
Need to add `orElse(null)` to the end of Line 116 or else the following if statement won't work.